### DISCUSSION TITLE
Using GitLens without GitHub sign in

### DISCUSSION THREAD

**Discussion by corwin-of-amber** [gitkraken/vscode-gitlens#1408] [Q&A]
This seems to be a recurring issue, e.g. https://github.com/eamodio/vscode-gitlens/issues/1233.

I just want the plain git history functionality, without logging in to GitHub. But I still have a nagging notification in the accounts icon on the bottom left. 😢 
<img width="346" alt="Screen Shot 2021-01-21 at 0 56 56" src="https://user-images.githubusercontent.com/704372/105251112-92ff9980-5b83-11eb-8178-82c1190893b5.png">

[This comment](https://github.com/eamodio/vscode-gitlens/issues/1233#issuecomment-739714771) suggests there is an easy opt-out in the Insiders edition. I am using that edition, but could not find this feature. I have disabled *all* the GitHub-related options in "GitLens: Open Settings", but the notification is still there.

**Comment by eamodio** [comment]
In VS Code 1.54, you to silence/dismiss those request indicators (since VS Code owns that whole experience), by choosing _Deny_ from the associated prompt. You can also if you want set `"gitlens.integrations.enabled": false` and that will disable all of GitLens' rich integrations, which will stop any permission requests. Though I would recommend choosing, _Deny_ on the prompt, rather than disabling all the integrations. As in the future, that setting could affect many more things.

**Reply by corwin-of-amber** [reply]
Thanks! That's what I thought would happen initially, but after installing the Insiders edition of GitLens, the prompt does not show at all, only the notification on the bottom left. I should have mentioned that I am using the Insiders version of VSCode itself (VSCode 1.55) as well as the GitLens Insiders (v2021.3.802).

**Reply by eamodio** [reply]
If you click on the "notification" under the _Accounts_ icon, then you will get the prompt.

**Reply by corwin-of-amber** [reply]
Do you mean, click on "Sign in to use GitLens (Insiders) (1)" in the popup menu?

I must be missing something basic, because as soon as I click that, I am redirected to the GitHub sign-in page in my browser.

(is this a macOS-only issue perhaps..?)

**Reply by corwin-of-amber** [reply]
Is it a problem with my local configuration then? Can you please show me where in the code this functionality is implemented so that I can better debug it?

**Reply by eamodio** [reply]
Sorry, I was mistaken. You only get that prompt if you have already authenticated with GitHub, but haven't yet given GitLens access. I've opened https://github.com/microsoft/vscode/issues/118668 for VS Code to address this, because this whole UX is owned by VS Code.

So until that gets resolved, the only way to avoid this notification is to set `"gitlens.integrations.enabled": false`

**Reply by corwin-of-amber** [reply]
Thanks for the update! Yes, I have disabled integrations for now, so the immediate problem is resolved.

**Reply by mjmatthiesen** [reply]
> 
> 
> You can also if you want set `"gitlens.integrations.enabled": false` and that will disable all of GitLens' rich integrations, which will stop any permission requests.

Can you detail what that integration option actually entails? Is it just for github integration? We use azure devops, so it doesn't really offer anything to me if that's the case. I've disabled it for now so I can get rid of the 1 on the account for "sign in to use gitlens". It's actually the first time I've seen it.
